cocos/asset/assets/simple-texture.ts

```diff
@@ -140,11 +141,11 @@ export class SimpleTexture extends TextureBase {
      * @param level @en Mipmap level to upload the image to. @zh 要上传的 mipmap 层级。
      * @param arrayIndex @en The array index. @zh 要上传的数组索引。
      */
-    public uploadData (source: HTMLCanvasElement | HTMLImageElement | ArrayBufferView | ImageBitmap, level = 0, arrayIndex = 0) {
+    public uploadData (source: HTMLCanvasElement | HTMLImageElement | ArrayBufferView | ImageBitmap | IMemoryImageSource, level = 0, arrayIndex = 0) {
```

Comment:
As the purpose is not use `HTMLCanvasElement `, `HTMLImageElement `, `ArrayBufferView ` and `ImageBitmap ` in engine, pal/image should handle it internally.

Reply:
I found there is ImageSource in pal/image/types, why not use it?

Reply:
This is an exported interface.' ImageSource' is not exported to the developer.

Reply:
But how to fix the issue that, engine internally don't use web standard class?

Reply:
The engine internal is to remove the dependencies, the export interface still needs to be retained.

Reply:
Why not export ImageSource? It will keep compatibility and remove the usage of web classes.